exercises/04-integration/after/Divergent.Customers/ProgramService.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.ComponentModel;
+using System.ServiceProcess;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Logging;
+using Divergent.Customers.Config;
+
+namespace Divergent.Customers
+{
+    [DesignerCategory("Code")]
+    class ProgramService : ServiceBase
+    {
+        IEndpointInstance endpoint;
+        static ILog logger;
+
+        static ProgramService()
+        {
+            //TODO: optionally choose a custom logging library
+            // https://docs.particular.net/nservicebus/logging/#custom-logging
+            // LogManager.Use<TheLoggingFactory>();
+            logger = LogManager.GetLogger<ProgramService>();
+        }
+
+        public static void Main()
```

Comment:
I'd prefer not to conflate `Program.Main` and the `ServiceBase`. Can we keep them separate?